orchagent/portsorch.cpp

```diff
@@ -7231,17 +7212,18 @@ void PortsOrch::refreshPortStateAutoNeg(const Port &port)
         return;
     }
 
-    string adv_speeds;
+    string adv_speeds = "N/A";
 
-    if (getPortAdvSpeeds(port, true, adv_speeds))
-    {
-        m_portStateTable.hset(port.m_alias, "rmt_adv_speeds", adv_speeds);
-    }
-    else
+    if (port.m_admin_state_up)
```

Comment:
Previously the get remote adv speeds is done even when admin state is down and this new logic will restrict it to only when admin up. 
This means polling becomes irrelevant. Can you explain the change for the reason?

Reply:
When the port is admin down, the SerDes is stopped, and it's impossible to get a correct remote advertisement, hence this redundant operation should be skipped if the port is disabled.

Reply:
Then why do we need the poller? Can't we just update the remote advertised speeds on link up? The entire polling logic was introduced to get remote advertisement when link was down. Should that be removed?

Reply:
Sorry for the confusion, but that's a different story

1. MAC disabled + link down
This is a case we have port.m_admin_state_up == false, while the link is surely down, and the remote advertisement is not available since the MAC/SerDes is disabled

2. MAC enabled + link down
This is a case we have port.m_admin_state_up == true, while the link is operationally down, and the remote advertisement could be available as the MAC/SerDes is enabled, frame exchanges are happening on the PHY layer (i.e. It's the internal PHY, in this case. Precisely speaking, it's the TSC drivers on the Broadcom chips)

3. MAC enabled + link up
This is a general case of port.m_admin_state_up == true and link is  operationally up.

Conclusion:
While the remote advertisement could be available in case 2 and 3, it's not available in case 1, hence the poller should only fetching the status when MAC/SerDes is enabled.